specs/DeployApp.groovy

```diff
@@ -0,0 +1,133 @@
+    static def params = [
+            configname               : configName,
+            wlstabspath              : getWlstPath(),
+            appname                  : 'sample',
+            apppath                  : "$directory/$filename",
+            targets                  : 'AdminServer',
+
+            is_library               : '',
+            stage_mode               : '',
+            plan_path                : '',
+            deployment_plan          : '',
+            overwrite_deployment_plan: '',
+            additional_options       : '',
+            archive_version          : '',
+            retire_gracefully        : '',
+            retire_timeout           : '',
+            version_identifier       : '',
+            upload                   : '',
+            remote                   : '',
+    ]
+
+    // TODO: check what environment is used
+    def static serverHostname = System.getProperty('COMMANDER_SERVER') ?: '10.200.1.150'
+
+
+    def doSetupSpec() {
+        createConfig(configName)
+
+        setupRepository(serverHostname)
+        setupResource(weblogicResource)
+
+        publishArtifact(artifactName, version, filename)
+
+        dslFile "dsl/procedures.dsl", [
+                projectName: projectName,
+                procedureName: procedureName,
+                resourceName: getResourceName(),
+                params: params
+        ]
+
+        //TODO: check why falling using this one
+        // This adds 120 seconds to test run time
+        try {
+            downloadArtifact(artifactName, directory, weblogicResource)
+        }
+        catch (Throwable e){
+            logger.debug("Job succeeds (and file is really downloaded) but test is failing")
+        }
+
+
+        // Check page procedure
+        def targetUrl = 'http://weblogic:7001/sample/hello.jsp'
+        dslFile "dsl/procedures.dsl", [
+                projectName: projectName,
+                subProjectName : '/plugins/EC-WebLogic/project',
+                procedureName: 'CheckPageStatus',
+                params: [
+                    successcriteria : 'pagefound',
+                    targeturl      : targetUrl
+                ]
+        ]
+    }
+
+    def 'Deploy application'() {
+        given:
+        def applicationName = 'sample'
+        def applicationWarPath = "$directory/$filename"
+        when:
+        def result = runProcedure("""
+        runProcedure(
+            projectName: '$projectName',
+            procedureName: '$procedureName',
+            actualParameter: [
+                 configname : '$configName',
+                 wlstabspath: '$wlstPath',
+                 appname : '$applicationName',
+                 apppath : "$applicationWarPath",
+                 targets : 'AdminServer',
+            ]
+        )
+        """, getResourceName())
+        then:
+        logger.debug(result.logs)
+        assert result.outcome == 'success'
+    }
+
+    def 'Check deployment'() {
```

Comment:
Please move this into a parent helper, we will need this method often. Create something akin to runWLST: checkPage(url) -> code, text

Reply:
Also: you cannot rely on the order of features, and you cannot make them dependent. If you want to test deployed application, do this in the "deploy application" feature.